### DISCUSSION TITLE
RelationManager hooks!

### DISCUSSION THREAD

**Discussion by ALSHIHAB** [filamentphp/filament#8658] [Help] [answered]
What are the hooks of RelationManager and how to use them?
I'm in a big trouble of using Filament. I spent about a month getting stuck with every step as there is no good documentation.

**Comment by ALSHIHAB** [comment, marked as answer]
I've got the answer after many tries!!.
Actually the form hooks started from table actions. This is what I'm looking for. Therefore, in table actions you can fire action hooks. for an example. in the table of RelationManger:
```
 ->headerActions([
                Tables\Actions\CreateAction::make()
                    ->mutateFormDataUsing(function (array $data): array {
                        $data['user_id'] = auth()->id();
                        // dd($data);
                        return $data;
                    })
                    ->beforeFormFilled(function () {

                        // Runs before the form fields are populated with their default values.
                    }),
            ])
```

**Comment by wychoong** [comment]
> I'm in a big trouble of using Filament. I spent about a month getting stuck with every step as there is no good documentation.

You can submit PR to improve the doc
